internal/virtualcircuit/update.go

```diff
@@ -0,0 +1,127 @@
+package virtualcircuit
+
+import (
+	"context"
+	"fmt"
+
+	metal "github.com/equinix-labs/metal-go/metal/v1"
+	"github.com/spf13/cobra"
+)
+
+func (c *Client) Update() *cobra.Command {
+	var (
+		description string
+		speed       string
+		name        string
+		vnid        string
+		vcID        string
+		subnet      string
+		customerIP  string
+		metalIP     string
+		md5         string
+		tags        []string
+		peerAsn     int
+	)
+	// updateConnectionCmd represents the updateConnectionCmd command
+	updateVirtualCircuitCmd := &cobra.Command{
+		Use:   `update -i <virtualcircuit_id> [--vnid <vn UUID>] [-d <description>] [-n <name>] [-s <speed>] [-t <tags>]`,
+		Short: "Updates a virtualcircuit.",
+		Long:  "Updates a specified virtualcircuit.",
+		Example: `  # Updates a specified virtualcircuit.:
+  metal virtual-circuit update --id 30c15082-a06e-4c43-bfc3-252616b46eba`,
+
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			inc := []string{}
+			var virtualCircuitUpdateInput metal.VirtualCircuitUpdateInput
+			header := []string{"ID", "Name", "Type", "Created"}
+
+			if vnid != "" {
+				vlanVirtualCircuitUpdateInput := metal.NewVlanVirtualCircuitUpdateInput()
+
+				if speed != "" {
+					vlanVirtualCircuitUpdateInput.SetSpeed(speed)
+				}
+
+				if description != "" {
+					vlanVirtualCircuitUpdateInput.SetDescription(description)
+				}
+
+				if name != "" {
+					vlanVirtualCircuitUpdateInput.SetName(name)
+				}
+
+				if len(tags) > 0 {
```

Comment:
It should also be possible to clear all tags but I don't know that we have precedent for that in metal-cli today.

Would I expect this command to add a tag or reset all tags:
* `metal thing update --tags foo // reset to one tag - foo?`
* `metal thing update --tags // clear all?`
* `metal thing update --tag foo // add a single tag foo? with no way to remove a single tag foo?`

I think there is more to consider on tag setting. Happy to consider that outside of the scope of this PR and simply document the behavior this command uses for now.

Reply:
> tags: Adds or updates the tags for the connection --tags="tag1,tag2".

In this case, I believe the behavior is actually "updates".  Tags are not being appended to the existing list, all existing tags are replaced with the specified tags. The documentation should match the behavior.